.github/workflows/codespell.yml

```diff
@@ -0,0 +1,30 @@
+# A Github action that using codespell to check spell.
+# .codespell/.codespellrc is a config file.
+# .codespell/wordlist.txt is a list of words that will ignore word checks.
+# More details please check the following link:
+# https://github.com/codespell-project/codespell
+
+name: Codespell
+
+on: pull_request
+
+jobs:
+  codespell:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: pip cache
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: ${{ runner.os }}-pip-
+
+      - name: Install prerequisites
+        run: sudo pip install -r ./.codespell/requirements.txt
+
+      - name: Spell check
+        run: codespell --config=./.codespell/.codespellrc
```

Comment:
This shouldn't be its own workflow. Please incorporate this functionality into the `zk` tool as I suggested